packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -109,6 +108,7 @@ export function normalizeAttributeValue(
     escapedExpression: boolean;
 } {
     const { name, value } = attr;
+    const [, rawValue] = raw.split('=');
```

Comment:
`rawValue` should be either `string` | `undefined`. Typescript inference is wrong here.